agent/main.py

```
@@ -165,15 +171,81 @@ def main():
+                # 宿主 UI 已消失。此刻 msg_thread 仍永久阻塞在 recv() 上
+                # (Transceiver 的 timeout_ 是 milliseconds::max())，调 shut_down()
+                # 会挂死在它内部的 join() 里，所以只能硬退出。
+                gone = watchdog.exited_target or f"pid={watchdog.ppid}"
+                mfaalog.error(f"[Agent] 宿主进程 {gone} 已退出，Agent 立即终止")
+                watchdog.close()
+                cleanup_socket_file(socket_id)
+                sys.stdout.flush()
+                sys.stderr.flush()
+                os._exit(1)
+    except KeyboardInterrupt:
+        # 与「宿主已退出」同一个死锁：能中断到这里，msg_thread 几乎必然还卡在 recv()，
```

Ruff (0.16.1)

[warning] Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)